### PR TITLE
MISUV-4517 Upgrade to Scala 2.13.x - income-tax-calculation

### PR DESCRIPTION
**Pull request by andriikondiuk**
### Description
Upgrade income-tax-calculation to Scala 2.13.8 

Add a link to the relevant story in Jira
[MISUV-4250](https://jira.tools.tax.service.gov.uk/browse/MISUV-4250)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: test/connectors/httpParsers/LiabilityCalculationHttpParserSpec.scala [hunk 1]
```diff
@@ -154,7 +154,7 @@ class LiabilityCalculationHttpParserSpec extends TestSuite {
 
         parser.CreateIncomeSourcesHttpReads.read("POST", "url", HttpResponse(OK, response)) mustBe
           Left(DesErrorModel(INTERNAL_SERVER_ERROR, DesErrorBodyModel("PARSING_ERROR",
-            "Error parsing response from DES - List((/id,List(JsonValidationError(List(error.path.missing),WrappedArray()))))")))
+            "Error parsing response from DES - List((/id,List(JsonValidationError(List(error.path.missing),List()))))")))
```
Comment (avelx) [inline test/connectors/httpParsers/LiabilityCalculationHttpParserSpec.scala:157]: List.empty to reduce number of brakets

**Review by avelx — APPROVED** [review]
looks good

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -23,18 +23,19 @@ object AppDependencies {
 
   val compile: Seq[ModuleID] = Seq(
     "uk.gov.hmrc"             %% "bootstrap-backend-play-28"  % bootstrapPlay28Version,
-    "uk.gov.hmrc.mongo"       %% "hmrc-mongo-play-28"         % "0.68.0",
-    "com.fasterxml.jackson.module" %% "jackson-module-scala"  % "2.12.2"
+    "uk.gov.hmrc.mongo"       %% "hmrc-mongo-play-28"         % "0.73.0",
+    "com.fasterxml.jackson.module" %% "jackson-module-scala"  % "2.14.0",
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply project/AppDependencies.scala:27]: It's a library version 2.14.0

File: test/connectors/httpParsers/LiabilityCalculationHttpParserSpec.scala (hunk 1 above)
Comment (andriikondiuk) [reply test/connectors/httpParsers/LiabilityCalculationHttpParserSpec.scala:157]: Unfortunately, it's not under our control it's what we will get as a String representation

Comment (avelx) [reply test/connectors/httpParsers/LiabilityCalculationHttpParserSpec.scala:157]: List() => List.empty ?

**Review by avelx — APPROVED** [review]
approved

**Pull request merged**